packages/sanity/src/hydration-test/event-listener-test.tsx

```diff
@@ -30,50 +30,56 @@ export const eventListenerTest = (): void => {
+           * React-supported event. This way, React will never call document.addEventListener during rendering of the components
+           * we want to test, and we can be sure that any remaining events are leftover.
+           */
+          before(() => {
+            ReactDOM.render(<AllEvents />, root);
+            ReactDOM.unmountComponentAtNode(root);
+          });
 
-        componentMetadata.simulations.forEach((simulation) => {
-          it('component should unmount without leaving event listeners on the window, document, and body', () => {
-            const windowLogger = attachEventListenerLogger(window);
-            const documentLogger = attachEventListenerLogger(document);
-            const bodyLogger = attachEventListenerLogger(document.body);
+          componentMetadata.simulations.forEach((simulation) => {
+            it('component should unmount without leaving event listeners on the window, document, and body', () => {
+              const windowLogger = attachEventListenerLogger(window);
+              const documentLogger = attachEventListenerLogger(document);
+              const bodyLogger = attachEventListenerLogger(document.body);
 
-            root.innerHTML = componentStrings[index];
-            hydrate(componentMetadata.simulationToJSX(simulation), root);
-            ReactDOM.unmountComponentAtNode(root);
-            index++;
+              root.innerHTML = componentStrings[index];
+              hydrate(componentMetadata.simulationToJSX(simulation), root);
+              ReactDOM.unmountComponentAtNode(root);
+              index++;
 
-            const errors = [
-              ...leftoverListenerErrors(windowLogger.listeners.getAll(), 'window'),
-              ...leftoverListenerErrors(documentLogger.listeners.getAll(), 'document'),
-              ...leftoverListenerErrors(bodyLogger.listeners.getAll(), 'body')
-            ];
+              const errors = [
+                ...leftoverListenerErrors(windowLogger.listeners.getAll(), 'window'),
+                ...leftoverListenerErrors(documentLogger.listeners.getAll(), 'document'),
+                ...leftoverListenerErrors(bodyLogger.listeners.getAll(), 'body')
+              ];
 
-            windowLogger.detach();
-            documentLogger.detach();
-            bodyLogger.detach();
+              windowLogger.detach();
+              documentLogger.detach();
+              bodyLogger.detach();
 
-            if (errors.length) {
-              throw Error(errors.join('\n'));
-            }
+              if (errors.length) {
+                throw Error(errors.join('\n'));
+              }
+            });
           });
         });
-      });
+      } else {
+        // tslint:disable-next-line:no-console
+        console.error(`Skipping event listener test for Component: "${componentName}". The "nonEventListenerCheckCompliant" flag was set to true in the metadata.`);
```

Comment:
why `console.error`. It isn't an error if someone meant for the test not to run no?